cumulusci/tasks/salesforce/tests/test_UpdateDependencies.py

```diff
@@ -49,11 +49,16 @@ def test_run_task(self, ApiRetrieveInstalledPackages):
         }
         task.api_class = mock.Mock()
         zf = zipfile.ZipFile(io.BytesIO(), "w")
-        with mock.patch.dict(
-            "cumulusci.tasks.salesforce.UpdateDependencies._install_dependency.__func__.__globals__",
-            download_extract_zip=mock.Mock(return_value=zf),
-        ):
+        try:
+            UD_globals = UpdateDependencies._install_dependency.__globals__
+        except AttributeError:  # Python 2
+            UD_globals = UpdateDependencies._install_dependency.__func__.__globals__
+        download_extract_zip = UD_globals["download_extract_zip"]
+        UD_globals["download_extract_zip"] = mock.Mock(return_value=zf)
```

Comment:
we should probably restructure the task so that we don't need to do this much state-maintenance in a test

Reply:
The real problem is that the salesforce package imports the task class from a module of the same name so there isn't a good way for us to get the module to patch its download_extract_zip global.